### PR TITLE
Add LocalDriver for Scheduler

### LINKED ISSUE #6759: Move LOCAL driver from the feature branch under the Scheduler module.
It is up for discussion, but since Scheduler requires new python driver it would be reasonable to move it under the scheduler module first.

### PR DESCRIPTION
**Pull request by pinkwah**
Resolves #6759

### CODE REVIEW

File: src/ert/scheduler/driver.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            preexec_fn=os.setpgrp,
+        )
+        await self.event_queue.put((iens, JobEvent.STARTED))
+        try:
+            if await proc.wait() == 0:
+                await self.event_queue.put((iens, JobEvent.COMPLETED))
+            else:
+                await self.event_queue.put((iens, JobEvent.FAILED))
+        except asyncio.CancelledError:
+            proc.kill()
```
Comment (xjules) [inline src/ert/scheduler/driver.py:71]: `proc.terminate()` ?

**Review by xjules — APPROVED** [review]
Nice! Let's bring it in :rocket:

File: src/ert/scheduler/driver.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    STARTED = 0
+    COMPLETED = 1
+    FAILED = 2
+    ABORTED = 3
+
+
+class Driver(ABC):
+    """Adapter for the HPC cluster."""
+
+    event_queue: asyncio.Queue[Tuple[int, JobEvent]]
```
Comment (xjules) [inline src/ert/scheduler/driver.py:22]: We need to do typing.Queue here:
```from typing import Queue
event_queue: Queue[Tuple[int, JobEvent]] ....
```

**Review by xjules — CHANGES_REQUESTED** [review]
Small update required with typing

File: src/ert/scheduler/driver.py (hunk 2 above)
Comment (pinkwah) [reply src/ert/scheduler/driver.py:24]: I think the solution is `from __future__ import annotations`

Comment (xjules) [reply src/ert/scheduler/driver.py:24]: Or that :)

**Review by xjules — APPROVED** [review]
Re-approved :+1:

**Pull request merged**